_posts/2020-06-28-openfaas-loki-into.md

```diff
@@ -0,0 +1,235 @@
+
+To have logs for long ago terminated function instances, we need to install some kind of logs aggregator. Traditionally, this has meant the [ELK stack](https://www.elastic.co/what-is/elk-stack): ElasticSearch, Logstash, Kibana. This is a great and well established tool; but, in [late 2018 Grafana introduced](https://kccna18.sched.com/event/GrXC/on-the-oss-path-to-full-observability-with-grafana-david-kaltschmidt-grafana-labs) a new alternative aimed squarely at the Cloud Native world: [Loki](https://grafana.com/oss/loki/).
+
+> Loki is a horizontally-scalable, highly-available, multi-tenant log aggregation system inspired by Prometheus. It is designed to be very cost effective and easy to operate. It does not index the contents of the logs, but rather a set of labels for each log stream.
+
+Loki is a light weight, easy to install, very easy to use. It is especially easy to use in Kuberenetes.
+
+Once a log aggregator is installed and configured it will often have a UI for exploring the logs, for ELK this is [Kibana](https://www.elastic.co/kibana) and for Loki it is [Grafana Explore](https://grafana.com/docs/grafana/latest/features/explore/). However, since OpenFaaS Gateway 0.15.0+ (July 2019), OpenFaaS supports [pluggable logging providers](https://docs.openfaas.com/architecture/logs-provider/) to support streaming function logs via the Gateway API and the `faas-cli`.
+
+The simplicy of Loki made it easy to turn into the first external [log provider](https://docs.openfaas.com/architecture/logs-provider/) for OpenFaaS that can now be easily installed using [arkade](https://get-arkade.dev/), which we will walk-though below.
+
+
+## Prerequisites
+
+Before we start we need a couple of tools to help us quickly set up our environment:
+
+* [docker](https://docs.docker.com/get-docker/) - the container runtime used in this post
+* [kubernetes](https://kind.sigs.k8s.io/docs/user/quick-start/) - cluster which will manage our containers
+* [arkade](https://github.com/alexellis/arkade) - one line installation of applications with which we will install OpenFaaS
+* [faas-cli](https://docs.openfaas.com/cli/install/) - the CLI which communicates with the OpenFaaS gateway
+* [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) - the CLI which communicates with your Kubernetes cluster
```

Comment:
I would like to see KinD listed here - add (installed below) if you don't want them to install it during the pre-reqs.